### ISSUE TITLE
Helix3d crashes when using WARP (software) rendering with custom shader

### ISSUE THREAD

**Issue by mjkkirschner** [helix-toolkit/helix-toolkit#1328]
Version: [2.9]
Package: [Helixtoolkit.Wpf.SharpDX]
Issue:

when using a custom shader as defined here:
https://github.com/DynamoDS/Dynamo/pull/10462

Helix throws a null reference exception inside `PhongMaterialVariables` constructor when my graphics adapter is disabled (WARP rendering)

![Screen Shot 2020-03-20 at 8 55 28 PM](https://user-images.githubusercontent.com/508936/77362308-16afe600-6d27-11ea-8adb-6c31af41493e.png)

please see the two hlsl files:
https://github.com/mjkkirschner/Dynamo/blob/1e2aed9424c76cc5c493637b0ea1fcedd874fed1/src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/psDynamoMesh.hlsl
https://github.com/mjkkirschner/Dynamo/blob/1e2aed9424c76cc5c493637b0ea1fcedd874fed1/src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/vsDynamoMesh.hlsl

these are compiled for vs_5 and ps_5 - 

if I recompile the shaders for vs_4 and ps_4 - no crash occurs, but we get this stippled effect:
![Screen Shot 2020-03-23 at 4 38 42 PM](https://user-images.githubusercontent.com/508936/77361030-dfd8d080-6d24-11ea-9f51-75c57ea32e18.png)

I am using `SV_IsFrontFace` - maybe it is related?

Reproduce Steps:
* disable graphics adapter through device manager
* compile shaders with compile shaders .bat also in PR
* add these shaders to effectsManager, and use associated DynamoMeshCore and DynamoMeshNode classes.

I have debugged and with the exception dialog in visual studio, and unfortunately it does not tell me much - maybe because of aggressive inlining that is occurring during this code path?
I have reverted my shader changes and the standard default shaders work fine, so it's clear it's something that is triggered by these shaders.

It appears the line number indicated is not actually the cause - that line refers to the depth pass - but looks like it should not be required as it retrieves a null pass if no pass with that name is found.

Sample Code:
see PR - if required I can try to modify the custom shader demo to replicate the issue.

Curious if the maintainers have any suggestions before digging into this further.

**Comment by holance** [comment]
It is always tricky to handle software rendering mode. You may have to check what instructions are supported in software rendering mode. Try to avoid using software rendering mode if possible. Otherwise it needs a lot of resources to test the compatibility issues.

**Comment by JeremyAnsel** [comment]
Hello,

`vs_5` and `ps_5` require feature level 11.
`vs_4` and `ps_4` require feature level 10.

By default, the WARP device was created using `FeatureLevel.Level_10_0`.
This is why the shaders that use `vs_5` or `ps_5` don't work.

In the pull request, I've modified how the WARP device is created.
I've removed the `FeatureLevel.Level_10_0` in the device creation code. This way the device is created with the highest available feature level.
